perun-core/src/test/java/cz/metacentrum/perun/core/entry/AttributesManagerEntryIntegrationTest.java

```diff
@@ -8744,10 +8874,4948 @@ public void setAttributeRights() throws Exception {
+	public void getResourceAttributesByGroup() throws Exception {
+		System.out.println(CLASS_NAME + "getResourceAttributesByGroup");
+
+		setAttributesForResourceAttributesTest();
+
+		Method testedMethod = getPrivateMethodFromAtrManager("getResourceAttributes",
+				PerunSession.class, Group.class, AttributeDefinition.class);
+
+		//find test resource attributes for group2InVo1
+		List<RichAttribute> ra_group2InVo1 = (List<RichAttribute>) testedMethod.invoke(attributesManagerBl,
+				sess, group2InVo1, resource_test_atr_def);
+		List<Attribute> attrs_group2InVo1 = new ArrayList<>();
+		ra_group2InVo1.forEach(ra -> attrs_group2InVo1.add(ra.getAttribute()));
+
+		assertEquals("Invalid number of attributes found", 2, attrs_group2InVo1.size());
+		assertTrue(attrs_group2InVo1.contains(resource1InVo1_test_atr));
+		assertTrue(attrs_group2InVo1.contains(resource2InVo1_test_atr));
+
+		//find test resource attributes for membersGroupOfVo1
+		List<RichAttribute> ra_membersGroupOfVo1 = (List<RichAttribute>) testedMethod.invoke(attributesManagerBl,
+				sess, membersGroupOfVo1, resource_test_atr_def);
+		List<Attribute> attrs_membersGroupOfVo1 = new ArrayList<>();
+		ra_membersGroupOfVo1.forEach(ra -> attrs_membersGroupOfVo1.add(ra.getAttribute()));
+
+		assertEquals("Invalid number of attributes found", 0, attrs_membersGroupOfVo1.size());
+	}
+
+	@SuppressWarnings({"unchecked"})
+	@Test
+	public void getResourceAttributesByResource() throws Exception {
+		System.out.println(CLASS_NAME + "getResourceAttributesByResource");
+
+		setAttributesForResourceAttributesTest();
+
+		Method testedMethod = getPrivateMethodFromAtrManager("getResourceAttributes",
+				PerunSession.class, Resource.class, AttributeDefinition.class);
+
+		//find test resource attributes for resource1InVo1
+		List<RichAttribute> ra_resource1InVo1 = (List<RichAttribute>) testedMethod.invoke(attributesManagerBl,
+				sess, resource1InVo1, resource_test_atr_def);
+		List<Attribute> attrs_resource1InVo1 = new ArrayList<>();
+		ra_resource1InVo1.forEach(ra -> attrs_resource1InVo1.add(ra.getAttribute()));
+
+		assertEquals("Invalid number of attributes found", 1, attrs_resource1InVo1.size());
+		assertTrue(attrs_resource1InVo1.contains(resource1InVo1_test_atr));
+	}
+
+	@SuppressWarnings({"unchecked"})
+	@Test
+	public void getResourceAttributesByVo() throws Exception {
+		System.out.println(CLASS_NAME + "getResourceAttributesByVo");
+
+		setAttributesForResourceAttributesTest();
+
+		Method testedMethod = getPrivateMethodFromAtrManager("getResourceAttributes",
+				PerunSession.class, Vo.class, AttributeDefinition.class);
+
+		//find test resource attributes for vo1
+		List<RichAttribute> ra_vo1 = (List<RichAttribute>) testedMethod.invoke(attributesManagerBl,
+				sess, vo1, resource_test_atr_def);
+		List<Attribute> attrs_vo1 = new ArrayList<>();
+		ra_vo1.forEach(ra -> attrs_vo1.add(ra.getAttribute()));
+
+		assertEquals("Invalid number of attributes found", 2, attrs_vo1.size());
+		assertTru
```

Comment:
I know it's like that used also somewhere else, but it makes test DB content dependent. We can and sometimes want run tests against normal DB. In such case getting ``entityless:def:namespace-minUID`` might fail. Because it's created only by sql for in-memory db.  You should catch ``AttributeNotExistsException`` and create Attribute in such case.

Or better think about usage in a test - if you really need THIS attribute to exists - e.g. because usage is hardcoded in attribute module of another attribute you are using. If no, create new test attribute from same namespace instead.

Same applies to user,member etc. attributes you are using in this ``setAttributesForXXXXTest()``. Only attributes present in AttributesManagerImpl init method should really exists in DB once perun-core is started.